src/code/code_800E8EA0.c

```diff
@@ -1,107 +1,138 @@
+/**
+ * Computes the necessary HeadRot and TorsoRot steps to be added to the normal rotation to smoothly turn an actors's
+ * head and torso towards the player if within a certain yaw.
+ *
+ * @param[in] globalCtx
+ * @param[in] actor
+ * @param[in,out] headRot the computed actors' head's rotation step
+ * @param[in,out] torsoRot the computed actors' torso's rotation step
+ * @param[in] focusPosYAdj how much to adjust the focus position's Y value from the actor's world position
+ *
+ * @return s32 boolean, true if rotated towards player, false if roatated back forwards.
+ *
+ * @note if in a cutscene or debug camera is enabled, the computed rotation will turn towards the view eye no matter the
+ * yaw.
+ */
+s32 Actor_TurnToPlayer(GlobalContext* globalCtx, Actor* actor, Vec3s* headRot, Vec3s* torsoRot, f32 focusPosYAdj) {
     Player* player = GET_PLAYER(globalCtx);
-    s16 sVar3;
-    Vec3f local_14;
+    s16 yaw;
+    Vec3f target;
 
     actor->focus.pos = actor->world.pos;
-    actor->focus.pos.y += param_5;
+    actor->focus.pos.y += focusPosYAdj;
 
-    if (((globalCtx->csCtx).state == 0) && (D_801D0D50 == 0)) {
-        sVar3 = ABS_ALT(BINANG_SUB(actor->yawTowardsPlayer, actor->shape.rot.y));
-        if (sVar3 >= 0x4300) {
-            func_800E8F08(param_3, param_4);
-            return 0;
+    if (!((globalCtx->csCtx.state != 0) || gDbgCamEnabled)) {
+        yaw = ABS_ALT(BINANG_SUB(actor->yawTowardsPlayer, actor->shape.rot.y));
+        if (yaw >= 0x4300) {
+            Actor_RotateForward(headRot, torsoRot);
+            return false;
         }
     }
 
-    if (((globalCtx->csCtx).state != 0) || (D_801D0D50 != 0)) {
-        local_14 = globalCtx->view.eye;
+    if ((globalCtx->csCtx.state != 0) || gDbgCamEnabled) {
+        target = globalCtx->view.eye;
     } else {
-        local_14 = player->actor.focus.pos;
+        target = player->actor.focus.pos;
     }
 
-    func_800E8FA4(actor, &local_14, param_3, param_4);
+    Actor_RotateToPoint(actor, &target, headRot, torsoRot);
 
-    return 1;
+    return true;
 }
 
-s32 func_800E9250(GlobalContext* globalCtx, Actor* actor, Vec3s* param_3, Vec3s* param_4, Vec3f param_5) {
+/**
+ * Computes the necessary HeadRot and TorsoRot steps to be added to the normal rotation to smoothly turn an actors's
+ * head and torso towards the player if within a certain yaw, else rotate back forward.
+ *
+ * @param[in] globalCtx
+ * @param[in] actor
+ * @param[in,out] headRot the computed actors' head's rotation step
+ * @param[in,out] torsoRot the computed actors' torso's rotation step
+ * @param[in] focusPos the point to set as the actor's focus position
+ *
+ * @return s32 boolean, true if rotated towards player, false if roatated back forwards.
+ *
+ * @note if in a cutscene or debug camera is enabled, the computed rotation will turn towards the view eye no matter the
+ * yaw.
+ */
+s32 Actor_TurnToPlayerSetFocus(GlobalContext* globalCtx, Actor* actor, Vec3s* headRot, Vec3s* torsoRot,
```

Comment:
It would be good to emphasise more in the comments what the difference is between these two.

Reply:
Added another sentence to each to help explain.